### PR TITLE
Feat/renew gacha system

### LINKED ISSUE #6: Two different gacha rolls by command
You should be able to launch two commands, each one should launch a gacha with different prize and cost item.

### PR DESCRIPTION
**Pull request by AdanJoGoHe**
Resolves #6

### CODE REVIEW

File: src/main/java/net/eriagacha/RegisterCommands.java [hunk 1]
```diff
@@ -45,8 +54,8 @@ public static void registerCommands() {
   }
 
 
-  public static void gachaRun(CommandContext<ServerCommandSource> ctx) throws CommandSyntaxException {
-    GachaController.giveItem(ctx);
+  public static void gachaRoll(CommandContext<ServerCommandSource> ctx, ItemStack moneyCondition) throws CommandSyntaxException {
+    GachaController.giveGachaReward(ctx, moneyCondition);
```
Comment (adelfuns) [inline src/main/java/net/eriagacha/RegisterCommands.java:58]: I would rename this method to giveReward

File: src/main/java/net/eriagacha/models/GachaObjectModelItem.java [hunk 2]
```diff
@@ -33,20 +33,23 @@ public void reward(CommandContext<ServerCommandSource> ctx) throws CommandSyntax
...
     GachaTelemetryController.InsertTelemetry(
         ctx.getSource().getName(),
         this.getItem().getTranslationKey()
     );
 
+
     ctx.getSource().sendFeedback(new LiteralText(
-        "Has obtenido " + this.getItemQuantity() + " " +
-            new TranslatableText(this.getItem().getTranslationKey()).getString()), false);
+        TextUtils.translatedTextToString("text.eriagacha.obtained")
```
Comment (adelfuns) [inline src/main/java/net/eriagacha/models/GachaObjectModelItem.java:47]: Use String.format

File: src/main/java/net/eriagacha/utils/TextUtils.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package net.eriagacha.utils;
+
+import net.minecraft.text.TranslatableText;
+
+public class TextUtils {
+
+  public static String translatedTextToString(String keyOfTranslatedText){
+
```
Comment (adelfuns) [inline src/main/java/net/eriagacha/utils/TextUtils.java:8]: This line break 👎

File: src/main/java/net/eriagacha/controller/GachaController.java [hunk 4]
```diff
@@ -86,19 +87,22 @@ public static void loadTheGacha() {
...
     boolean moneyConditionsMet =
-        ctx.getSource().getPlayer().getInventory().contains(GachaUtils.GACHA_REQUIEREMENT);
+        ctx.getSource().getPlayer().getInventory().contains(moneyCondition);
 
     if (!moneyConditionsMet) {
-      throw new SimpleCommandExceptionType(
-          new TranslatableText("Te falta una " + GachaUtils.GACHA_REQUIEREMENT
-              .getTranslationKey())).create();
+      String missing = new TranslatableText("text.eriagacha.missing").getString();
+      String MoneyCondition = new TranslatableText(moneyCondition.getTranslationKey()).getString();
```
Comment (adelfuns) [inline src/main/java/net/eriagacha/controller/GachaController.java:98]: MoneyCondition -> moneyCondition

File: src/main/java/net/eriagacha/controller/GachaController.java [hunk 5]
```diff
@@ -86,19 +87,22 @@ public static void loadTheGacha() {
...
+      throw new SimpleCommandExceptionType(new LiteralText(missing+" "+MoneyCondition)).create();
```
Comment (adelfuns) [inline src/main/java/net/eriagacha/controller/GachaController.java:99]: String.format

**Review by adelfuns — CHANGES_REQUESTED** [review]

File: src/main/java/net/eriagacha/RegisterCommands.java (hunk 1 above)
Comment (AdanJoGoHe) [reply src/main/java/net/eriagacha/RegisterCommands.java:58]: Done

File: src/main/java/net/eriagacha/models/GachaObjectModelItem.java (hunk 2 above)
Comment (AdanJoGoHe) [reply src/main/java/net/eriagacha/models/GachaObjectModelItem.java]: Done

File: src/main/java/net/eriagacha/controller/GachaController.java (hunk 4 above)
Comment (AdanJoGoHe) [reply src/main/java/net/eriagacha/controller/GachaController.java:98]: Done

File: src/main/java/net/eriagacha/utils/TextUtils.java (hunk 3 above)
Comment (AdanJoGoHe) [reply src/main/java/net/eriagacha/utils/TextUtils.java]: Done

File: src/main/java/net/eriagacha/controller/GachaController.java (hunk 5 above)
Comment (AdanJoGoHe) [reply src/main/java/net/eriagacha/controller/GachaController.java]: Done

**Review by adelfuns — APPROVED** [review]

**Pull request merged**